File: dataset/human/java/file_1287.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();

        long total = 0;
        long a[] = new long[n];
        for (int i = 0; i < n; i++) {
            a[i] = in.nextInt();
            total += a[i];
        }

        long former = 0;
        long min = Long.MAX_VALUE;
        for (int i = 0; i < n - 1; i++) {
            former += a[i];
            min = Math.min(min, Math.abs(total - 2 * former));
        }

        System.out.println(min);
    }
}

File: dataset/human/python/file_1287.py
import itertools

N = int(input())
A = list(map(int,input().split()))
total = sum(A)
ACC = list(itertools.accumulate(A))

ans = 10 ** 10
for i in range(N-1):
  x = ACC[i]
  y = total - x
  ans = min(ans,abs(x - y))

print(ans)
